runtime/rastrace/method_trace.c

```diff
@@ -202,15 +201,43 @@ traceMethodEnter(J9VMThread *thr, J9Method *method, void *receiverAddress, UDATA
 			} else {
 				Trc_MethodEntry(thr, J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(methodName), J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodSignature), J9UTF8_DATA(methodSignature), receiver);
 			}
+		}
+	}
 
-			if (doParameters) {
-				Trc_MethodArguments(
-						thr, 
-						(U_32)J9UTF8_LENGTH(receiverClassName),
-						J9UTF8_DATA(receiverClassName),
-						receiver,
-						traceMethodArguments(thr, methodSignature, (UDATA*)receiverAddress - 1, buf, buf + sizeof(buf)));
+	if (0 != doParameters) {
```

Comment:
This should be nested within the else block (not a compiled method); only the receiver is available for compiled, non-static methods.